woocommerce/google-listings-and-ads
# Display the budget currency that will actually be used.

### Changes proposed in this Pull Request:

- Request budget recommendation in Ads account's currency,
instead of store's currency. The user provides the amount in Ads currency which is fixed and unchangeable, therefore the recommendation should be given in the same one.

Solves part of https://github.com/woocommerce/google-listings-and-ads/issues/363#issuecomment-883081178


### Screenshots:
##### Shop currency
![image](https://user-images.githubusercontent.com/17435/126819061-301917e6-88f2-468d-8bc6-372db46d63e2.png)

#### Ads currency 
![image](https://user-images.githubusercontent.com/17435/126819728-19a08916-69e1-497d-b189-28cd28aa2bb0.png)

##### Budget recommendation currency
![image](https://user-images.githubusercontent.com/17435/126818769-8e1e80b2-42f9-4afb-bacc-0999ce6023f2.png)
![currency](https://user-images.githubusercontent.com/17435/126979341-1093fbe6-819d-48a8-a850-b6d6fdcc711f.gif)




### Detailed test instructions:

1. Set the store currency to _X_ (PLN)
2. Setup Ads account
3. Change the store currency to _Y_ (EUR)
4. Create a new campaign ([`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Fcampaigns%2Fcreate`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Fcampaigns%2Fcreate) )
5. You should see the budget input and recommendation requested and displayed in the Ads account currency _X_ - the one that will be used after clicking "Launch paid campaign". The one visible at https://ads.google.com/aw/campaigns? 


### Changelog entry

> Fix - display the correct currency actually used for the paid campaign budget.



### Additional notes:
1. Even though the Ads currency code is displayed, the ** store's currency formatting settings** (thousand & decimal delimiters, etc.) are being used.
2. We end up with the blink of no country code in the input. This is due to the number of requests we make for displaying the page. Browsers limit the number of concurrent requests, so this one gets significantly delayed. @j111q is that acceptable? We may try to find some ways to work around it by pre-loading this data before displaying the countries list.
3. There still are a number of things to improve around Ads currency, I'll address them in separate PRs:
	1. Show a warning that the store and Ads currencies differ.
	2. Fix the currency displayed in the Dashboard.
